lib/ansible/modules/cloud/azure/azure_rm_trafficmanagerprofile.py

```diff
@@ -0,0 +1,451 @@
+                type='str',
+                default='present',
+                choices=['present', 'absent']
+            ),
+            location=dict(
+                type='str'
+            ),
+            profile_status=dict(
+                type='str',
+                default='enabled',
+                choices=['enabled', 'disabled']
+            ),
+            routing_method=dict(
+                type='str',
+                default='performance',
+                choices=['performance', 'priority', 'weighted', 'geographic']
+            ),
+            dns_config=dict(
+                type='dict',
+                options=dns_config_spec
+            ),
+            monitor_config=dict(
+                type='dict',
+                default=dict(
+                    protocol='HTTP',
+                    port=80,
+                    path='/'
+                ),
+                options=monitor_config_spec
+            ),
+        )
+
+        self.resource_group = None
+        self.name = None
+        self.state = None
+        self.tags = None
+        self.location = None
+        self.profile_status = None
+        self.routing_method = None
+        self.dns_config = None
+        self.monitor_config = None
+        self.endpoints_copy = None
+
+        self.results = dict(
+            changed=False
+        )
+
+        super(AzureRMTrafficManagerProfile, self).__init__(derived_arg_spec=self.module_arg_spec, supports_check_mode=True)
+
+    def exec_module(self, **kwargs):
+
+        for key in list(self.module_arg_spec.keys()) + ['tags']:
+            setattr(self, key, kwargs[key])
+
+        to_be_updated = False
+
+        if not self.dns_config:
+            self.dns_config = dict(
+                relative_name=self.name,
+                ttl=60
+            )
+
+        if not self.location:
+            self.location = 'global'
+
+        response = self.get_traffic_manager_profile()
+
+        if self.state == 'present':
+            if not response:
+                to_be_updated = True
+            else:
+                self.results = response
+                self.log('Results : {0}'.format(response))
+                update_tags, response['tags'] = self.update_tags(response['tags'])
+
+                if update_tags:
+                    to_be_updated = True
+
+                to_be_updated = to_be_updated or self.check_update(response)
+
+            if to_be_updated:
+                self.log("Need to Create / Update the Traffic Manager profile")
+
+                if not self.check_mode:
+                    self.results = shorten_traffic_manager_dict(self.ceate_update_traffic_manager_profile())
+                    self.log("Creation / Update done.")
+
+                self.results['changed'] = True
+                return self.results
+
+        elif self.state == 'absent' and response:
+            self.log("Need to delete the Traffic Manager profile")
+            self.results['changed'] = True
```

Comment:
results should also contain the id and endpoints ids of the resource that was deleted. There's not too much point in knowing this post delete but it keeps the return values more consistent.